mlrun/serving/remote.py

```diff
@@ -124,25 +172,34 @@ async def _handle_completed(self, event, response):
 
         body = self._get_data(response_body, response.headers)
 
-        if body is not None:
-            new_event = self._user_fn_output_to_event(event, body)
-            await self._do_downstream(new_event)
+        new_event = self._user_fn_output_to_event(event, body)
+        await self._do_downstream(new_event)
 
     def do_event(self, event):
         # sync implementation (without storey)
         if not self._session:
             self._session = requests.Session()
+            http_adapter = get_http_adapter(self.retries, self.backoff_factor)
             self._session.mount("http://", http_adapter)
             self._session.mount("https://", http_adapter)
 
         body = _extract_input_data(self._input_path, event.body)
         method, url, headers, body = self._generate_request(event, body)
         try:
             resp = self._session.request(
-                method, url, verify=False, headers=headers, data=body
+                method,
+                url,
+                verify=False,
+                headers=headers,
+                data=body,
+                timeout=self.timeout,
+            )
+        except requests.exceptions.ReadTimeout as err:
```

Comment:
Are you sure this is the only relevant timeout? Why not the more general `requests.exceptions.Timeout` which also covers connection timeouts?